mlos_core/mlos_core/optimizers/flaml_optimizer.py

```diff
@@ -26,35 +26,47 @@ class EvaluatedSample(NamedTuple):
 
 
 class FlamlOptimizer(BaseOptimizer):
-    """Wrapper class for FLAML Optimizer: A fast library for AutoML and tuning.
-
-    Parameters
-    ----------
-    parameter_space : ConfigSpace.ConfigurationSpace
-        The parameter space to optimize.
-
-    space_adapter : BaseSpaceAdapter
-        The space adapter class to employ for parameter space transformations.
-
-    low_cost_partial_config : dict
-        A dictionary from a subset of controlled dimensions to the initial low-cost values.
-        More info: https://microsoft.github.io/FLAML/docs/FAQ#about-low_cost_partial_config-in-tune
-
-    seed : Optional[int]
-        If provided, calls np.random.seed() with the provided value to set the seed globally at init.
+    """
+    Wrapper class for FLAML Optimizer: A fast library for AutoML and tuning.
     """
 
     def __init__(self, *,
                  parameter_space: ConfigSpace.ConfigurationSpace,
+                 optimization_targets: List[str],
                  space_adapter: Optional[BaseSpaceAdapter] = None,
                  low_cost_partial_config: Optional[dict] = None,
                  seed: Optional[int] = None):
+        """
+        Create an MLOS wrapper class for FLAML.
 
+        Parameters
+        ----------
+        parameter_space : ConfigSpace.ConfigurationSpace
+            The parameter space to optimize.
+
+        optimization_targets : List[str]
+            The names of the optimization targets to minimize.
+            For FLAML it must be a list with a single element, e.g., `["score"]`.
+
+        space_adapter : BaseSpaceAdapter
+            The space adapter class to employ for parameter space transformations.
+
+        low_cost_partial_config : dict
+            A dictionary from a subset of controlled dimensions to the initial low-cost values.
+            More info: https://microsoft.github.io/FLAML/docs/FAQ#about-low_cost_partial_config-in-tune
+
+        seed : Optional[int]
+            If provided, calls np.random.seed() with the provided value to set the seed globally at init.
+        """
         super().__init__(
             parameter_space=parameter_space,
+            optimization_targets=optimization_targets,
             space_adapter=space_adapter,
         )
 
+        if len(self._optimization_targets) != 1:
+            raise ValueError("FLAML does not support multi-target optimization")
```

Comment:
Hmm, this seems to indicate that multi-objective is supported?  Maybe we could reach out to Chi and discuss further.  Could also adjust the message for now and extend support for FLAML in a future PR.
https://microsoft.github.io/FLAML/docs/reference/tune/searcher/suggestion#optunasearch-objects

Reply:
Actually, maybe it's just optuna that does that.  Not sure if we need a wrapper for a wrapper.  Maybe we could "just" incorporate optuna directly?

https://optuna.readthedocs.io/en/stable/index.html
https://optuna.readthedocs.io/en/stable/tutorial/10_key_features/002_configurations.html

Reply:
Oh, I thought SMAC did something a little different actually, but it's been a while since I looked.
I know that when Johannes chatted with them at NeurIPS they said the initial implementation was not the best and they were working on improving it.

Reply:
Just checked the SMAC3 code again. By default, they do the weighted average of the targets: [aggregation_strategy.py:44](https://github.com/automl/SMAC3/blob/main/smac/multi_objective/aggregation_strategy.py#L44)
There's also an implementation of [ParEGO](https://www.cs.bham.ac.uk/~jdk/UKCI-2015.pdf), but it is not used by default.

It would be pretty easy to implement weighted average aggregation strategy on top of FLAML; also, maybe we should pass the target weights as an optional parameter to all our optimizers?

Reply:
Ah, ParEGO was the one I was thinking of.  Johannes had done some experiments with that.